### PR TITLE
live-preview: Delay updaing the preview

### PR DESCRIPTION
**Pull request by hunger**
Delay updating the live preview to prevent a
thundering herd effect when e.g. a git update changes a lot of files at once.

### CODE REVIEW

File: tools/lsp/preview.rs [hunk 1]
```diff
@@ -1049,33 +1051,8 @@ pub fn reload_preview() {
...
-                cache.loading_state = PreviewFutureState::NeedsReload;
-                return;
-            }
-            PreviewFutureState::NeedsReload => return,
-        }
-        cache.loading_state = PreviewFutureState::PreLoading;
-    };
-
-    let result = run_in_ui_thread(move || async move {
+fn reload_timer_function() -> impl std::future::Future<Output = ()> {
```
Comment (ogoffart) [inline tools/lsp/preview.rs:1054]: Why not make this function async?

**Review by ogoffart — COMMENTED** [review]
i don't understand why this is a problem in the first place, this is why we had loading_state, and if it is pre-loading, it should just return.

File: tools/lsp/preview.rs (hunk 1 above)
Comment (hunger) [reply tools/lsp/preview.rs:1054]: Yeah... why not? I guess confusion :-) I tried a lot till I arrived here :-)

Comment (hunger) [reply tools/lsp/preview.rs]: Resolved in latest push.

**Comment by hunger** [comment]
The filesystem notifications do not come in in one batch, but get happen with a bit of delay between them. I had that delay be enough for the first rendering to be done and then trigger a rerendering. We did never get a full one-invalidation == one-render-run, but I did get up to three re-renderings when playing around with bigger customer data. Typically it rendered twice.

**Pull request merged**